pkg/storage/interfaces.go

```diff
@@ -167,4 +167,6 @@ type Interface interface {
 	//    }
 	// })
 	GuaranteedUpdate(ctx context.Context, key string, ptrToType runtime.Object, ignoreNotFound bool, precondtions *Preconditions, tryUpdate UpdateFunc) error
+
+	Xxx(ctx context.Context, key string, out runtime.Object, ignoreNotFound bool, precondtions *Preconditions, tryUpdate UpdateFunc, guess runtime.Object) error
```

Comment:
> in the fullness of time, GuaranteedUpdate should be the only method in Cacher (the Xxx should only be in storage-specific implementation if at all).

If this is the case, yes, we should do the key fetching at cacher layer instead of GU.

The expected workflow is:

1. try update with fetched key info from cache
2. fallback to normal GU if 1 fails

Right?